### PR TITLE
Minor cleanups in core

### PR DESCRIPTION
**Pull request by oscargus**
#### Brief description of what is fixed or changed
Fixed some minor code issues (duplicate imports, unused variables, etc)

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/core/expr.py [hunk 1]
```diff
@@ -3619,7 +3616,7 @@ def _expand_hint_key(hint):
             use_hint = hints[hint]
             if use_hint:
                 hint = '_eval_expand_' + hint
-                expr, hit = Expr._expand_hint(expr, hint, deep=deep, **hints)
+                expr = Expr._expand_hint(expr, hint, deep=deep, **hints)[0]
```
Comment (oscarbenjamin) [inline sympy/core/expr.py:3619]: I consider this kind of thing to be defensive programming. If the function is always expected to return a sequence of length 2 then this asserts that it has exactly two items:
```python
>>> a, b = [1, 2, 3]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: too many values to unpack (expected 2)
```
The variable name can also be used to help understand what the function returns even if the `hit` variable is unused.

File: sympy/core/mul.py [hunk 2]
```diff
@@ -1113,7 +1113,7 @@ def _matches_noncomm(nodes, targets, repl_dict={}):
     def _matches_add_wildcard(dictionary, state):
         node_ind, target_ind = state
         if node_ind in dictionary:
-            begin, end = dictionary[node_ind]
+            begin = dictionary[node_ind][0]
```
Comment (oscarbenjamin) [inline sympy/core/mul.py:1116]: I wouldn't change this.

File: sympy/core/expr.py (hunk 1 above)
Comment (oscargus) [reply sympy/core/expr.py:3619]: I was also considering this and, e.g., for the mp-function returning unused accuracy variables I did not change it. I see the point to some extent here as well. (And next comment.)

File: sympy/core/mul.py (hunk 2 above)
Comment (smichr) [reply sympy/core/mul.py:1116]: But here and above you could use `_` for the unused arg.

Comment (smichr) [inline sympy/core/mul.py:1116]:
```suggestion
            begin, _ = dictionary[node_ind]
```

File: sympy/core/expr.py (hunk 1 above)
Comment (smichr) [inline sympy/core/expr.py:3619]:
```suggestion
                expr, _ = Expr._expand_hint(expr, hint, deep=deep, **hints)
```

File: sympy/core/mul.py (hunk 2 above)
Comment (oscarbenjamin) [reply sympy/core/mul.py]: I prefer `end`: it shows that the dictionary value is a 2-tuple marking the beginning and end which is helpful information when reading the code.

**Comment by oscargus** [comment]
Thanks @smichr . I changed them in my local copy (back to using a name, since it may make sense here) and also removed some mutable default arguments as I found #19372 and #20068. Hence, I disabled auto-merge.

**Comment by oscargus** [comment]
There is a bit of `repl_dict.copy()` here which I do not fullt know are required or if they were just a way to avoid mutating the default argument. I've kept them though.

**Comment by oscargus** [comment]
@smichr You are right! I had an idea at some stage, but then I automatically replaced all `if ...` with `if ... else .copy` and then I forgot this. Thanks!

**Pull request merged**